overlord/devicestate/handlers_systems.go

```diff
@@ -50,7 +51,7 @@ func taskRecoverySystemSetup(t *state.Task) (*recoverySystemSetup, error) {
 	if err == nil {
 		return &setup, nil
 	}
-	if err != state.ErrNoState {
+	if err != nil && !errors.Is(err, state.ErrNoState) {
```

Comment:
We know that `err` is not `nil` here, because it's checked a couple of lines above.

Reply:
You're right, thanks!